internal/process/queue.go

```diff
@@ -15,78 +17,111 @@ type Executor interface {
 }
 
 type Queue struct {
-	queue     workqueue.RateLimitingInterface
-	executor  Executor
-	waitGroup sync.WaitGroup
-	log       logrus.FieldLogger
+	queue                    workqueue.RateLimitingInterface
+	executor                 Executor
+	waitGroup                sync.WaitGroup
+	log                      logrus.FieldLogger
+	name                     string
+	workerExecutionTimes     map[string]time.Time
+	warnAfterTime            time.Duration
+	healthCheckFrequencyTime time.Duration
 
 	speedFactor int64
 }
 
-func NewQueue(executor Executor, log logrus.FieldLogger) *Queue {
+func NewQueue(executor Executor, log logrus.FieldLogger, name string, warnAfterTime, healthCheckFrequencyTime time.Duration) *Queue {
+	// add queue name field that could be logged later on
 	return &Queue{
-		queue:     workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "operations"),
-		executor:  executor,
-		waitGroup: sync.WaitGroup{},
-		log:       log,
-
-		speedFactor: 1,
+		queue:                    workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "operations"),
+		executor:                 executor,
+		waitGroup:                sync.WaitGroup{},
+		log:                      log.WithField("queueName", name),
+		speedFactor:              1,
+		name:                     name,
+		workerExecutionTimes:     make(map[string]time.Time),
+		warnAfterTime:            warnAfterTime,
+		healthCheckFrequencyTime: healthCheckFrequencyTime,
 	}
 }
 
 func (q *Queue) Add(processId string) {
 	q.queue.Add(processId)
+	q.log.Infof("added item %s to the queue %s, queue length is %d", processId, q.name, q.queue.Len())
 }
 
 func (q *Queue) AddAfter(processId string, duration time.Duration) {
 	q.queue.AddAfter(processId, duration)
+	q.log.Infof("item %s will be added to the queue %s after duration of %d, queue length is %d", processId, q.name, duration, q.queue.Len())
 }
 
 func (q *Queue) ShutDown() {
+	q.log.Infof("shutting down the queue, queue length is %d", q.queue.Len())
 	q.queue.ShutDown()
 }
 
 func (q *Queue) Run(stop <-chan struct{}, workersAmount int) {
+	q.log.Infof("starting workers, queue length is %d", q.queue.Len())
```

Comment:
Could we log workersAmount? When starting the queue length would be 0 unless we allow to insert elements before start.